api/controllers/UserController.js

```diff
@@ -100,8 +100,36 @@ module.exports = {
   },
 
   logout: function (req, res) {
-    delete req.session.usr;
-    req.session.loggedIn = false;
-    return res.ok();
+    // If the user isn't logged in, just get them out of here
+    if (!req.session.loggedIn) {
+      return res.ok();
+    }
+    // https://github.com/expressjs/session#sessiondestroycallback
```

Comment:
Finally found it-- it's an express session package, https://github.com/expressjs/session#sessiondestroycallback.

Reply:
This doesn't appear to be a promise but we _could_ theoretically call the underlying function that `session.js` uses, https://github.com/expressjs/session/blob/master/session/session.js#L101-L113.

Something like this?
```
const destroyUserSession = (req, res) => {
    return new Promise((resolve, reject) => {
        // Might need to be more defensive
        if (!req.session.id) {
            return reject('User not logged in');
        }
        req.session.destroy(function afterDestroy() {
            resolve('User logged out');
        });
    });
}

//

logout: async function (req, res) {
    await destroyUserSession(req, res);
    return res.ok();
},
```

Though I don't see a need to do it this way, we just avoided the abstraction.

Reply:
Cool good to know; thanks for digging. I agree; no point in abstracting it into a helper just to await` here